### PR TITLE
Allow omnibus to build "bundled" Windows installers

### PR DESCRIPTION
**Pull request by thommay**
The delivery-cli package needs to include the MS VC Redist libraries, and so we need to create a bundled installer.
This pull-request implements the logic to do so, and I'd appreciate a review of the approach before I go through and write the necessary tests and so on.

@jaym @ksubrama @schisamo

### CODE REVIEW

**Comment by sethvargo** [comment]
This is a bit out of my element, but the code looks good :+1:

**Comment by jaym** [comment]
@thommay why do we want both an msi and a bundle. Is it not pick one?

**Comment by jaym** [comment]
a few unit tests would be nice. Other than that :+1: 
Also, if you have a sample bundle.wxs file, i think it would be good to drop here

**Comment by thommay** [comment]
@jaym you build the msi, and then you chain the MSI into the exe

**Comment by schisamo** [comment]
:+1: Looks useful @thommay!

File: lib/omnibus/packagers/msi.rb [hunk 1]
```diff
@@ -88,6 +91,38 @@ class Packager::MSI < Packager::Base
...
+          bundle_file = windows_safe_path(Config.package_dir, bundle_name)
+
+          bundle_light_command = <<-EOH.split.join(' ').squeeze(' ').strip
+          light.exe
+            -nologo
+            -ext WixUIExtension
+            #{wix_extension_switches(wix_light_extensions)}
+            -cultures:en-us
+            -loc "#{windows_safe_path(staging_dir, 'localization-en-us.wxl')}"
+            project-files.wixobj bundle.wixobj
```
Comment (ksubrama) [inline lib/omnibus/packagers/msi.rb]: Would a bundle need the set of all project-files again?  I would have thought that the chaining process would only need some kind of reference to the msi.

**Comment by ksubrama** [comment]
http://wixtoolset.org/documentation/manual/v3/bundle/ seems to talk about a bootstrapper application and some new extension to be included.  Can you link to an example bundle.wxs file?  I'm just having a hard time figuring out what gets pulled into what.

**Comment by thommay** [comment]
@ksubrama yeah, previously I was just enabling the extension in the project file, but we can just turn it on across the board. A sample bundle.wxs is https://github.com/chef/delivery-cli/blob/tm/msi_build/omnibus-delivery-cli/resources/delivery-cli/msi/bundle.wxs.erb

**Comment by thommay** [comment]
@jaym @ksubrama @schisamo unit tests and a sample bundle.wxs added. If this looks good I'll squash it up

**Comment by schisamo** [comment]
@thommay This looks great! I'm :+1: after you squash things.

**Comment by danielsdeleo** [comment]
LGTM

**Comment by thommay** [comment]
@schisamo squished!

**Comment by schisamo** [comment]
@thommay Rebase and merge this!

**Pull request merged**